plugins/@grouparoo/sqlite/__tests__/table-import/schedule-options.ts

```diff
@@ -0,0 +1,54 @@
+import "@grouparoo/spec-helper";
+
+import { beforeData, afterData, getConfig } from "../utils/data";
+
+import { getConnection } from "../../src/lib/table-import/connection";
+const scheduleOptions = getConnection().scheduleOptions;
+
+// these used and set by test
+const { appOptions, purchasesTableName } = getConfig();
+const sourceOptions = { table: purchasesTableName };
+let client;
+
+async function getColumns() {
+  const columnOption = scheduleOptions[0];
+  const optionMethod = columnOption.options;
+
+  const response = await optionMethod({
+    connection: client,
+    appOptions,
+    sourceOptions,
+    app: null,
+    appId: null,
+    source: null,
+    sourceId: null,
+    sourceMapping: null,
+    properties: [],
+  });
+  return response;
+}
+
+describe("sqlite/table/scheduleOptions", () => {
+  beforeAll(async () => {
+    ({ client } = await beforeData());
+  });
+
+  afterAll(async () => {
+    await afterData();
+  });
+  test("gets list of columns that can handle highwatermark", async () => {
+    const columns = await getColumns();
+    const columnNames = columns.map((r) => r.key).sort();
+    // In SQLite, data is essentially either text or a number, so they could all
+    // technically be HWM columns.
```

Comment:
It looks like you have chosen to filter out strings at least, which is a good step to making this list possible HWMs shorter.

Reply:
It's actually not filtering anything out. It's just that this list is coming from the purchases data, so there aren't as many columns.